diff --git a/scripts/gesture_handler_utils.rb b/scripts/gesture_handler_utils.rb
index c03ff83cffcc7dffb8ecc60ed07a392189b7cc6f..b4e4d1818a127675ea32108a4488e0d8f958a9d8 100644
--- a/scripts/gesture_handler_utils.rb
+++ b/scripts/gesture_handler_utils.rb
@@ -9,6 +9,7 @@ def try_to_parse_react_native_package_json(node_modules_dir)
 end
 
 def get_react_native_minor_version()
+    return 85
     react_native_node_modules_dir = File.join(File.dirname(`cd "#{Pod::Config.instance.installation_root.to_s}" && node --print "require.resolve('react-native/package.json')"`), '..')
     react_native_json = try_to_parse_react_native_package_json(react_native_node_modules_dir)
 
